source/extensions/filters/http/jwt_authn/verifier.cc

```diff
@@ -48,11 +48,21 @@ class ContextImpl : public Verifier::Context {
   // Stores an authenticator object for this request.
   void storeAuth(AuthenticatorPtr&& auth) { auths_.emplace_back(std::move(auth)); }
 
+  // Add a pair of (issuer, payload), called by Authenticator
+  void addPaylod(const std::string& issuer, const std::string& payload) {
+    payload_pairs_.push_back({issuer, payload});
+  }
+
+  bool hasPayload() const { return !payload_pairs_.empty(); }
+
+  ProtobufWkt::Struct getPayload() const { return MessageUtil::stringPairStruct(payload_pairs_); }
+
 private:
   Http::HeaderMap& headers_;
   Verifier::Callbacks* callback_;
   std::unordered_map<const Verifier*, CompletionState> completion_states_;
   std::vector<AuthenticatorPtr> auths_;
+  std::vector<std::pair<std::string, std::string>> payload_pairs_;
```

Comment:
Just use ProbotufWkt::Struct here?

Reply:
I am looking for a utility code to convert JSON to protobuf::Struct.  Do you know any?

Reply:
How about just use MessageUtil::loadFromJson() to convert it to Struct.   Not sure what it will look like?